x-pack/platform/plugins/shared/stack_alerts/public/rule_types/es_query/expression/esql_query_expression.tsx

```diff
@@ -25,20 +27,57 @@ import {
   firstFieldOption,
   getTimeFieldOptions,
   getTimeOptions,
+  isPerRowAggregation,
   parseAggregationResults,
 } from '@kbn/triggers-actions-ui-plugin/public/common';
 import { EsQueryRuleParams, EsQueryRuleMetaData, SearchType } from '../types';
 import { DEFAULT_VALUES, SERVERLESS_DEFAULT_VALUES } from '../constants';
 import { useTriggerUiActionServices } from '../util';
 import { hasExpressionValidationErrors } from '../validation';
 import { TestQueryRow } from '../test_query_row';
-import { rowToDocument, toEsQueryHits, transformDatatableToEsqlTable } from '../../../../common';
+import {
+  transformDatatableToEsqlTable,
+  getEsqlQueryHits,
+  ALERT_ID_SUGGESTED_MAX,
+} from '../../../../common';
+
+const ALL_DOCUMENTS = 'all';
+const alertingOptions = [
+  {
+    id: ALL_DOCUMENTS,
+    label: i18n.translate('xpack.stackAlerts.esQuery.ui.allDocumentsLabel', {
+      defaultMessage: 'Create an alert if matches are found',
+    }),
+  },
+  {
+    id: 'row',
+    label: i18n.translate('xpack.stackAlerts.esQuery.ui.alertPerRowLabel', {
+      defaultMessage: 'Create an alert for each row',
+    }),
+  },
+];
+
+const getWarning = (duplicateAlertIds: Set<string>, longAlertIds: Set<string>) => {
+  if (duplicateAlertIds.size > 0) {
+    return i18n.translate('xpack.stackAlerts.esQuery.ui.alertPerRowWarning', {
+      defaultMessage:
+        'Your alerts do not appear to be unique, we noticed multiple rows with the same alert ID.',
+    });
+  } else if (longAlertIds.size > 0) {
+    return i18n.translate('xpack.stackAlerts.esQuery.ui.alertPerRowAlertIdWarning', {
+      defaultMessage: 'Your alert IDs are made up of at least {max} fields.',
```

Comment:
I would add something actionable here too. `Reduce the number of fields used to create Alert ID. Suggested number 10` (That's really bad, don't use that verbatim 🙈 )

Reply:
Updated in this commit [4ecd058](https://github.com/elastic/kibana/pull/212135/commits/4ecd058fdcd74d4e2f0fb777de79587933aba71b)